onnxruntime/core/providers/migraphx/migraphx_provider_factory.cc

```diff
@@ -131,10 +134,7 @@ struct MIGraphX_Provider : Provider {
 
     migx_options.migraphx_use_native_calibration_table = internal_options.int8_use_native_calibration_table;
 
-    migx_options.migraphx_save_compiled_model = internal_options.save_compiled_model;
-    migx_options.migraphx_save_model_path = internal_options.save_model_file.c_str();
-    migx_options.migraphx_load_compiled_model = internal_options.load_compiled_model;
-    migx_options.migraphx_load_model_path = internal_options.load_model_file.c_str();
+    migx_options.migraphx_cache_dir = internal_options.model_cache_dir.string().c_str();
```

Comment:
.string().c_str() returns a temporary pointer I believe.  You need to store the result of internal_options.model_cache_dir.string() at somewhere.